web-app/src/pages/Dashboard/container.js

```diff
@@ -79,9 +79,19 @@ const DashboardContainer = Wrapped =>
       this.setState({ activeEmployee: employeeId }, this.filterCalenderEvents);
     };
 
+    handleCalendarNavigate = date => {
+      const newDate = new moment(date);
+      this.setState(
+        {
+          calendarDate: newDate.startOf('month').format('YYYY-MM-DD'),
```

Comment:
Is calendarDate not already set to be that `new moment().startOf('month').format('YYYY-MM-DD')` within the constructor so you don't need to do it within handleCalendarNavigate. Just assuming here, I could be wrong.

Reply:
`calenderDate` is initially set to `new moment().startOf('month').format('YYYY-MM-DD')`, but here we're updating it using `date` which comes from the calendar component, `date` is a javascript object rather than a moment object so we transform it into a moment object (`newDate`) which means we need to do the date modification and formatting again.

In short, we're setting state from a completely different object, so we need to transform it again.

Reply:
Oops didn't see the start of it! Sounds good!